python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1775 @@
+        band_label = exposure.filter.bandLabel
+
+        self.log.info("Extracted detector info: name='%s', camera='%s', band='%s'",
+                     detector_name, camera_name, band_label)
+
+        # Initialize data arrays
+        donut_data = {}
+
+        # Get positions from donut stamps - use standard column names and units
+        if hasattr(donutStamps, 'getSkyPositions'):
+            sky_positions = donutStamps.getSkyPositions()
+            donut_data["coord_ra"] = [
+                pos.getRa().asRadians() for pos in sky_positions
+            ]  # Use radians like standard
+            donut_data["coord_dec"] = [
+                pos.getDec().asRadians() for pos in sky_positions
+            ]  # Use radians like standard
+        else:
+            # Fallback: use exposure center
+            bbox = exposure.getBBox()
+            center_pos = exposure.wcs.pixelToSky(bbox.getCenterX(), bbox.getCenterY())
+            donut_data["coord_ra"] = [center_pos.getRa().asRadians()] * num_donuts
+            donut_data["coord_dec"] = [center_pos.getDec().asRadians()] * num_donuts
+
+        # Get centroid positions from TARTS centers (most accurate)
+        if hasattr(self.tarts, 'centers') and self.tarts.centers is not None:
+            self.log.debug(
+                "TARTS centers type: %s, shape: %s",
+                type(self.tarts.centers), getattr(self.tarts.centers, 'shape', 'no shape')
+            )
+
+            # Handle different data types for TARTS centers
+            if hasattr(self.tarts.centers, 'cpu'):
+                # PyTorch tensor
+                centers_array = self.tarts.centers.cpu().numpy()
+                self.log.info("TARTS centers numpy array shape: %s", centers_array.shape)
+            else:
+                # Already numpy array or list
+                centers_array = np.array(self.tarts.centers)
+                self.log.info("TARTS centers converted to numpy, shape: %s", centers_array.shape)
+
+            # Handle different array shapes
+            if centers_array.ndim == 1:
+                # Flattened array: [x1, y1, x2, y2, ...] -> reshape to [n, 2]
+                if len(centers_array) % 2 == 0:
+                    centers_array = centers_array.reshape(-1, 2)
+                    self.log.info("Reshaped flattened centers to shape: %s", centers_array.shape)
+                else:
+                    self.log.warning("TARTS centers length %d is not divisible by 2", len(centers_array))
+                    centers_array = None
+            elif centers_array.ndim == 2:
+                # Already in [n, 2] format
+                if centers_array.shape[1] != 2:
+                    self.log.warning("TARTS centers shape %s doesn't have 2 columns", centers_array.shape)
+                    centers_array = None
+            else:
+                self.log.warning("Unexpected TARTS centers dimensionality: %d", centers_array.ndim)
+                centers_array = None
```

Comment:
I can't tell if some of these should be `RuntimeError`s or `ValueError`s but I wouldn't be surprised.